tests/src/ApiClientTest.php

```diff
@@ -61,4 +61,35 @@ public function testToken(): void
         $client->setToken($token);
         $this->assertEquals($token, $client->getToken());
     }
+
+    /**
+     * @covers ::extractConfigValues
+     */
+    public function testExtractConfigValues(): void
+    {
+        $keys_to_extract = [
+            'existing_key',
+            'non_existing_key',
+            0,
+            '99',
+        ];
+
+        $client = $this->getTestingClient([
+            'existing_key' => 'Existing Key',
+            'other_key' => 'Other Key',
+            'boolean_value_key' => false,
+            0 => 'Zero',
+            '99' => 'Bottles',
```

Comment:
array_flip only supports numbers and strings as keys but I didnt add null and boolean as phpunit 10 does not support expectWarning anymore, think its sufficient coverage, but let me know if you think otherwise

Reply:
I think this is fully ok.